linode/manager
# M3-5610: Storybook for Radios

## Description

- Updates Storybook for the Radio component
- Now uses mdx
- Fixed padding on radio causing ripple to be a non-square size
- Use makeStyles() in the`<Radio />` component
- Removed the `error` and `warning` variant

## How to test

- Check for regressions with `<Radio />`
- View all Radio stories in Storybook (`yarn storybook`)

## 🔎 **Preview**
[https://m3-5610-update-story-for-rad.manager-ec2.pages.dev](https://m3-5610-update-story-for-rad.manager-ec2.pages.dev/?path=/docs/ui-elements-radio--radio#radio)

## Args Notes

It would be nice if Storybook automatically knew **all** of the types we generally extend in our own components (in this case all ButtonProps) so we don't have to manually populate the args. Seems like we're not the only ones having this issue. Typescript isn't inferring all prop types, so our args table is not completely populated with all possible props form MUI that can be used on the component 😕

[This fix](https://storybook.js.org/docs/ember/configure/typescript#overriding-the-configuration-to-infer-additional-props) did not seem to work on my initial attempt 

https://github.com/storybookjs/storybook/issues/12558
https://github.com/reactjs/react-docgen/issues/416
https://github.com/storybookjs/storybook/tree/next/addons/docs/react#typescript-props-with-react-docgen
